SubmissionHistory fixes

TomNUSDS opened the pull request:
- Do not display failed submissions in list.
 - SubmissionsResource differs from API result. Fix
 - Minor refactor of SubmissionTable to remove dup code and improve "undefined" checking.
 - ReportResource differs from API result. Fix

## Pull reviewers stats
Stats for the last 30 days:
|                                                                                                                                                                            | User               | Total reviews | Median time to review                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | Total comments |
| -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ | ------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------- |
| <a href="https://github.com/MauriceReeves-usds"><img src="https://avatars.githubusercontent.com/u/74194189?u=f81b3f069b28469e1fb39322ba3c17112c1dd1d0&v=4" width="20"></a> | MauriceReeves-usds | **32**        | [13h 56m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74194189~n~'MauriceReeves-usds)~p~30~r~(~(d~'r5i0jc~t~'d9)~(d~'r5m3yg~t~'3ljr)~(d~'r5nlxx~t~'vlw)~(d~'r5hxz6~t~'1qv)~(d~'r5jurt~t~'19l)~(d~'r58oe5~t~'8n)~(d~'r5pds7~t~'12k)~(d~'r5nmg4~t~'5cgu)~(d~'r6bk48~t~'8g)~(d~'r6bjqe~t~'16iu)~(d~'r62a4v~t~'zxg)~(d~'r68026~t~'3pc9)~(d~'r6dec9~t~'17ov)~(d~'r6bjqx~t~'1b6d)~(d~'r5ynan~t~'20dl)~(d~'r60l83~t~'18hg)~(d~'r5yp3j~t~'99vx)~(d~'r5yop9~t~'1d1o)~(d~'r5yn3y~t~'1tyq)~(d~'r67z1h~t~'7)~(d~'r5hy8t~t~'4wc6)~(d~'r60l0y~t~'1q)~(d~'r60l73~t~'2c2)~(d~'r5yu2x~t~'1whn)~(d~'r60kw1~t~'1bpm)~(d~'r60rbh~t~'da)~(d~'r684pw~t~'i)~(d~'r67yoj~t~'4vz3)~(d~'r682b4~t~'h)~(d~'r682xy~t~'aw)~(d~'r5yn0w~t~'1rcp)~(d~'r6bkdq~t~'7s)~(d~'r6bkly~t~'9x)~(d~'r6bjra~t~'1kkq)~(d~'r6bjo2~t~'15fl)~(d~'r6c1hp~t~'ae)~(d~'r6bldq~t~'12h)~(d~'r67zgg~t~'12g2q)))) | 11             |
| <a href="https://github.com/cwinters-usds"><img src="https://avatars.githubusercontent.com/u/86251310?u=e7f46bd48ca250dea43f9e0d6433a7fb8627ed30&v=4" width="20"></a>      | cwinters-usds      | 21            | [13h 48m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86251310~n~'cwinters-usds)~p~30~r~(~(d~'r5idk5~t~'l0)~(d~'r5ifa8~t~'dr)~(d~'r5awx0~t~'9bh)~(d~'r5ikyp~t~'9l)~(d~'r5il2o~t~'ozb)~(d~'r5lu9b~t~'8sts)~(d~'r5luav~t~'8svc)~(d~'r5lueo~t~'8sz5)~(d~'r56yxt~t~'1el)~(d~'r5akvq~t~'bn)~(d~'r5wnss~t~'8rb7)~(d~'r5wo3e~t~'8y92)~(d~'r5wo4m~t~'8yaa)~(d~'r5wo6q~t~'8yce)~(d~'r5wxin~t~'16r)~(d~'r58rbg~t~'1wb8)~(d~'r60fh9~t~'1fnz)~(d~'r60mwv~t~'4e)~(d~'r60txe~t~'4jj)~(d~'r60cux~t~'axnb)~(d~'r60cw0~t~'axoe)~(d~'r5yzk3~t~'3my)~(d~'r5wnq8~t~'c7d1)~(d~'r69wtd~t~'3td)~(d~'r6bs6m~t~'kq)~(d~'r681p4~t~'12ibe))))                                                                                                                                                                                                                                       | 6              |
| <a href="https://github.com/carlosfelix2"><img src="https://avatars.githubusercontent.com/u/63804190?u=009d26bf9914817d2f9a066811541d0f6f3155c9&v=4" width="20"></a>       | carlosfelix2       | 16            | [2h 58m](https://app.flowwer.dev/charts/review-time/~(u~(i~'63804190~n~'carlosfelix2)~p~30~r~(~(d~'r5m8hs~t~'ob)~(d~'r5m8hy~t~'oh)~(d~'r5nmxi~t~'103)~(d~'r5iew2~t~'jb)~(d~'r5b1qf~t~'e4w)~(d~'r5b1s4~t~'e6l)~(d~'r5b25c~t~'ejt)~(d~'r5b3n0~t~'g1h)~(d~'r58rzs~t~'1hrl)~(d~'r5b4ed~t~'2pp)~(d~'r5b4o2~t~'2ze)~(d~'r5cj3p~t~'19w5)~(d~'r5jv0z~t~'1ir)~(d~'r5jv6h~t~'85k)~(d~'r5jv9j~t~'88m)~(d~'r5ls88~t~'17xv)~(d~'r5nvd8~t~'1hip)~(d~'r5nvh1~t~'1hmi)~(d~'r5nw93~t~'1iek)~(d~'r5nwd6~t~'1iin)~(d~'r67wdj~t~'3lnm)~(d~'r6bmn4~t~'3wj)~(d~'r6bmq3~t~'3zi)~(d~'r6bn8c~t~'4hr)~(d~'r6bnap~t~'4k4)~(d~'r6fssg~t~'9p)~(d~'r5z012~t~'7s)~(d~'r5yoqk~t~'21ti)~(d~'r60ydi~t~'cg)~(d~'r60ydu~t~'cs)~(d~'r5pfqs~t~'17qs)~(d~'r5wvhb~t~'5n5m)~(d~'r5ww0q~t~'5np1)~(d~'r5x6n8~t~'91y)~(d~'r69t4b~t~'4b)~(d~'r69xot~t~'6ns)~(d~'r6drh4~t~'1zv8))))                           | **63**         |
| <a href="https://github.com/brick-green-agile6"><img src="https://avatars.githubusercontent.com/u/86254221?u=894ef5a4773dd01e92be595af8f1879041f85002&v=4" width="20"></a> | brick-green-agile6 | 15            | [1h 10m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86254221~n~'brick-green-agile6)~p~30~r~(~(d~'r5kdkf~t~'402)~(d~'r5ls19~t~'1mj)~(d~'r5wuiw~t~'n1)~(d~'r5pgvb~t~'39d)~(d~'r6btti~t~'1sn)~(d~'r6c0m4~t~'8l9)~(d~'r6fdj9~t~'y5)~(d~'r6fewv~t~'1b2)~(d~'r6e5ts~t~'2rd)~(d~'r56ww1~t~'1vt)~(d~'r5cs23~t~'40e)~(d~'r5csqs~t~'e8)~(d~'r5x9sg~t~'df67)~(d~'r62go4~t~'6g8)~(d~'r62ifd~t~'5v)~(d~'r5kdyd~t~'73m)~(d~'r62gt0~t~'1urw)~(d~'r67zwf~t~'4x6z)~(d~'r6801i~t~'52i0)~(d~'r6brw6~t~'1ar)~(d~'r6ab9n~t~'c33))))                                                                                                                                                                                                                                                                                                                                          | 3              |
| <a href="https://github.com/jimduff-usds"><img src="https://avatars.githubusercontent.com/u/62258514?u=b7ce81d1478c472fb5bf6f3d31edb1a4454d55dd&v=4" width="20"></a>       | jimduff-usds       | 14            | [13h 12m](https://app.flowwer.dev/charts/review-time/~(u~(i~'62258514~n~'jimduff-usds)~p~30~r~(~(d~'r5lq7w~t~'8w6y)~(d~'r5igom~t~'5f8t)~(d~'r5jrit~t~'12v5)~(d~'r5o6xg~t~'cuwf)~(d~'r57g4n~t~'1uc)~(d~'r58o52~t~'14ks)~(d~'r5ai4q~t~'1mga)~(d~'r5eki9~t~'1n9f)~(d~'r628oy~t~'yhj)~(d~'r6c63l~t~'2ol)~(d~'r60m2h~t~'nv)~(d~'r62ect~t~'10k)~(d~'r6a3l2~t~'ck1)~(d~'r6k3cb~t~'bim))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | 2              |
| <a href="https://github.com/TomNUSDS"><img src="https://avatars.githubusercontent.com/u/74203452?u=6eccca154e9d5c808c5269d9604648aec3c9a412&v=4" width="20"></a>           | TomNUSDS           | 14            | [1h 10m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74203452~n~'TomNUSDS)~p~30~r~(~(d~'r5kihi~t~'2h0)~(d~'r5m0z4~t~'1kym)~(d~'r5i85t~t~'5f7i)~(d~'r5lvf8~t~'3d0j)~(d~'r5lvmd~t~'3d7o)~(d~'r5lvyj~t~'3dju)~(d~'r5i74c~t~'31w)~(d~'r5i7b1~t~'38l)~(d~'r5ib9d~t~'4q)~(d~'r5cyi1~t~'l8)~(d~'r5nsa9~t~'4ay)~(d~'r5poww~t~'1g2)~(d~'r5pqr4~t~'1eg)~(d~'r5x4de~t~'12x)~(d~'r5pri0~t~'3wd7)~(d~'r6fugu~t~'3gg)~(d~'r5yx53~t~'2z)~(d~'r5ysqs~t~'1zlk)~(d~'r5z1i7~t~'3x5)~(d~'r612b0~t~'f3)~(d~'r610jw~t~'123))))                                                                                                                                                                                                                                                                                                                                                 | 16             |
| <a href="https://github.com/jbiskie"><img src="https://avatars.githubusercontent.com/u/84460447?v=4" width="20"></a>                                                       | jbiskie            | 9             | [1h 14m](https://app.flowwer.dev/charts/review-time/~(u~(i~'84460447~n~'jbiskie)~p~30~r~(~(d~'r55fqk~t~'rr)~(d~'r5m7ao~t~'si)~(d~'r5ikju~t~'7ol)~(d~'r5k73p~t~'6j)~(d~'r5i8ld~t~'6ia)~(d~'r5iorm~t~'92)~(d~'r5q0km~t~'36f)~(d~'r6fnmw~t~'1kf5)~(d~'r6fvqz~t~'3po)~(d~'r6fyt9~t~'2jm)~(d~'r68237~t~'54jp)~(d~'r69x8n~t~'4sk))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | 0              |
| <a href="https://github.com/sean-usds"><img src="https://avatars.githubusercontent.com/u/87096393?u=90127e88c4ff845b6a754d10151b10a5158562ff&v=4" width="20"></a>          | sean-usds          | 9             | [13m](https://app.flowwer.dev/charts/review-time/~(u~(i~'87096393~n~'sean-usds)~p~30~r~(~(d~'r5ic36~t~'mb)~(d~'r5ic4z~t~'o4)~(d~'r5lwb1~t~'1exp)~(d~'r5oa7x~t~'1t)~(d~'r5oa8j~t~'2f)~(d~'r5cz5e~t~'fv)~(d~'r5d2xw~t~'1a)~(d~'r5x42o~t~'s7)~(d~'r60yt6~t~'9o4)~(d~'r5pm6i~t~'55tb)~(d~'r5csi4~t~'9s))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | 3              |
| <a href="https://github.com/kevinhaube"><img src="https://avatars.githubusercontent.com/u/4022304?u=c0f80e2ad03386621394d1830dc33b7b5cc23680&v=4" width="20"></a>          | kevinhaube         | 7             | [8h 31m](https://app.flowwer.dev/charts/review-time/~(u~(i~'4022304~n~'kevinhaube)~p~30~r~(~(d~'r5lqhv~t~'1ahd)~(d~'r5lqj0~t~'1aii)~(d~'r6fmur~t~'35o1)~(d~'r6fsbl~t~'ak)~(d~'r6fru0~t~'tm)~(d~'r612bd~t~'fg)~(d~'r62hs2~t~'1ia9)~(d~'r6duug~t~'57))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | 1              |
| <a href="https://github.com/JosiahSiegel"><img src="https://avatars.githubusercontent.com/u/5522990?v=4" width="20"></a>                                                   | JosiahSiegel       | 4             | [33m](https://app.flowwer.dev/charts/review-time/~(u~(i~'5522990~n~'JosiahSiegel)~p~30~r~(~(d~'r5m6nq~t~'1e)~(d~'r5m0bp~t~'18p)~(d~'r5pgq9~t~'1in)~(d~'r5x1pp~t~'61i)~(d~'r682hf~t~'3xx))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | 2              |
| <a href="https://github.com/oslynn"><img src="https://avatars.githubusercontent.com/u/20068283?u=bcad2a3d00ffe648463e3fbe1894762aeab72404&v=4" width="20"></a>             | oslynn             | 4             | [20h 19m](https://app.flowwer.dev/charts/review-time/~(u~(i~'20068283~n~'oslynn)~p~30~r~(~(d~'r5aofo~t~'1gne)~(d~'r5rcmh~t~'1q31)~(d~'r6273k~t~'1o8j)~(d~'r6dn5p~t~'ws))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | 19             |
| <a href="https://github.com/jorg3lopez"><img src="https://avatars.githubusercontent.com/u/49923512?v=4" width="20"></a>                                                    | jorg3lopez         | 4             | [1h 6m](https://app.flowwer.dev/charts/review-time/~(u~(i~'49923512~n~'jorg3lopez)~p~30~r~(~(d~'r5i7xa~t~'4qe)~(d~'r56yw7~t~'1cz)~(d~'r5if0f~t~'box)~(d~'r62j5e~t~'ef))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | 1              |
| <a href="https://github.com/sethdarragile6"><img src="https://avatars.githubusercontent.com/u/92405130?u=b36435069928f7e5f03109c9f478cdc7158fc01e&v=4" width="20"></a>     | sethdarragile6     | 4             | [**6m**](https://app.flowwer.dev/charts/review-time/~(u~(i~'92405130~n~'sethdarragile6)~p~30~r~(~(d~'r5m4pt~t~'2h)~(d~'r5oall~t~'mma)~(d~'r612ap~t~'es)~(d~'r6duue~t~'55))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | 0              |
| <a href="https://github.com/RickHawesUSDS"><img src="https://avatars.githubusercontent.com/u/48692522?u=323d76e6f867d8ed0b3e56d7e5808313bfcc2ccf&v=4" width="20"></a>      | RickHawesUSDS      | 3             | [4h 6m](https://app.flowwer.dev/charts/review-time/~(u~(i~'48692522~n~'RickHawesUSDS)~p~30~r~(~(d~'r5i5qi~t~'1n9)~(d~'r5nzx9~t~'cnw8)~(d~'r6k385~t~'beg))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | 0              |
| <a href="https://github.com/rhood23699"><img src="https://avatars.githubusercontent.com/u/86322826?v=4" width="20"></a>                                                    | rhood23699         | 3             | [1h 57m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86322826~n~'rhood23699)~p~30~r~(~(d~'r6biq1~t~'2xn2)~(d~'r5xbox~t~'1c)~(d~'r69xv2~t~'5ez))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | 0              |
| <a href="https://github.com/clediggins-usds"><img src="https://avatars.githubusercontent.com/u/89804087?u=5b03415b2bc7766f96908dd40f522633187e4b02&v=4" width="20"></a>    | clediggins-usds    | 3             | [1h 36m](https://app.flowwer.dev/charts/review-time/~(u~(i~'89804087~n~'clediggins-usds)~p~30~r~(~(d~'r5i5lc~t~'4fa)~(d~'r53o24~t~'dyu)~(d~'r6jude~t~'2jp))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | 0              |
| <a href="https://github.com/ahay-agile6"><img src="https://avatars.githubusercontent.com/u/19178435?u=aa7d4ddd31665f83e677e20967502557741345e4&v=4" width="20"></a>        | ahay-agile6        | 2             | [1d 22h 26m](https://app.flowwer.dev/charts/review-time/~(u~(i~'19178435~n~'ahay-agile6)~p~30~r~(~(d~'r5lz7z~t~'3gta)~(d~'r5pkaz~t~'3p66))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | 0              |
| <a href="https://github.com/whytheplatypus"><img src="https://avatars.githubusercontent.com/u/410846?v=4" width="20"></a>                                                  | whytheplatypus     | 1             | [1d 2h 22m](https://app.flowwer.dev/charts/review-time/~(u~(i~'410846~n~'whytheplatypus)~p~30~r~(~(d~'r57dn1~t~'3eu)~(d~'r59bgs~t~'218l)~(d~'r59bip~t~'21ai)~(d~'r5cjuv~t~'f6)~(d~'r5i0kt~t~'4z50))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | 4              |
| <a href="https://github.com/jeremy-page"><img src="https://avatars.githubusercontent.com/u/54326889?u=600f8a60859e9b0ca4634227f00b2cd9bee5e9b8&v=4" width="20"></a>        | jeremy-page        | 1             | [21h 35m](https://app.flowwer.dev/charts/review-time/~(u~(i~'54326889~n~'jeremy-page)~p~30~r~(~(d~'r5pdtd~t~'1nz1))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | 0              |
| <a href="https://github.com/victorialb-usds"><img src="https://avatars.githubusercontent.com/u/92449888?u=7b96445c1d1f5bb2dc6b7d26fbe1ad170fbf510d&v=4" width="20"></a>    | victorialb-usds    | 1             | [2d 20h 38m](https://app.flowwer.dev/charts/review-time/~(u~(i~'92449888~n~'victorialb-usds)~p~30~r~(~(d~'r5pr0v~t~'5ano))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | 1              |
| <a href="https://github.com/jh765"><img src="https://avatars.githubusercontent.com/u/96186164?v=4" width="20"></a>                                                         | jh765              | 1             | [2h 54m](https://app.flowwer.dev/charts/review-time/~(u~(i~'96186164~n~'jh765)~p~30~r~(~(d~'r6dxgl~t~'821))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | 0              |

sethdarragile6 reviewed (changes requested):
Looking great, except I found a bug when hitting the Previous for the "ignore" org on staging (as I've let you know

TomNUSDS commented:
Also fixes #4144 (wrote up for regression testing)
Also, did some minor layout fixes. Ends up that changing the list length (to remove failed items), makes the Next and Previous button move around too much by putting them at the top and bottom of the table.

@sethdarragile6 @kevinhaube Can you re-review when you get a chance?

sethdarragile6 reviewed (approved):
@TomNUSDS Previous works now and all look great. Approved!

TomNUSDS commented:
Full set of changes before squash:
* Remove failed submissions from list.
* SubmissionHistory Bug fixes
* Next -> Next -> Next -> Prev -> Next  now works correctly.
* ReportResource differs from API result. Fix
* SubmissionsResource differs from API result. Fix
* Minor refactor of SubmissionTable to remove dup code and improve "undefined" checking.
* Fix page order when going prev
* createdAt is a string. Fix all code that assumed it was a Date()
* Now that it's working correctly, it seems like caching is working as expected.
* UI layout issues
* Because the list length now changes sizes if items are removed, the Next & Prev buttons squirrel around.  Added two copies of both buttons to top/bottom of table.
- Because the `paginationPageSize` is effectively a constant, I made it an actual constant.
* Made Next Previous links 508 compliant buttons.
*The logic around if there were any entries or if there were more than a single page of results confused me. I tried to make it more explicit.
* Refactored so all the `use....()` hooks were not at the top of the file.
